plugins/generator-addon-1.21.x/addon-1.21.x/templates/resourcepack/terrain_texture.json.ftl

```diff
@@ -4,14 +4,14 @@
 	"padding": 8,
 	"num_mip_levels": 0,
 	"texture_data": {
-		<#list w.getElementsOfType("block") as mod>
+		<#list w.getElementsOfType("beblock") as mod>
 			<#assign ge = mod.getGeneratableElement()>
-    		"${modid}_${mod.getRegistryName()}_up": { "textures": [ "textures/blocks/${ge.textureTop()}" ] },
+    		"${modid}_${mod.getRegistryName()}_up": { "textures": [ "textures/blocks/${ge.textureTop}" ] },

```

Comment:
You should keep those as methods, because those handle automatic fallback to "main" texture, which is a system I am not fond of removing, as it is quite convenient

Reply:
This is what I tried at the beginning, but they were causing errors, so I thought they were no longer supported. I'll have to search a bit to find the cause then.